File: backend/ruby_foundations/exercises/small_problems_101/easy_5/letter_swap.rb
def swap(sentence)
  new_sentence = sentence.split(' ').each do |word|
    temp = word[0]
    word[0] = word[-1]
    word[-1] = temp
  end
  new_sentence.join(' ')
end

puts swap('Oh what a wonderful day it is') == 'hO thaw a londerfuw yad ti si'
puts swap('Abcde') == 'ebcdA'
puts swap('a') == 'a'

File: frontend/javascript/exercises/easy_3/letter_swap.js
function swap(str) {
  words = str.split(' ');
  return words.map(function (word) {
    var wordArr = word.split('');
    var temp = wordArr[0];
    wordArr[0] = wordArr[word.length - 1];
    wordArr[word.length - 1] = temp;
    return wordArr.join('');
  }).join(' ');
}

console.log(swap('Oh what a wonderful day it is'));    // 'hO thaw a londerfuw yad ti si'
console.log(swap('Abcde'));                            // 'ebcdA'
console.log(swap('a'));                                // 'a'